orchagent/aclorch.cpp

```diff
@@ -2447,10 +2464,123 @@ void AclOrch::doTask(Consumer &consumer)
     }
 }
 
-bool AclOrch::addAclTable(AclTable &newTable, string table_id)
+void AclOrch::getAddDeletePorts(AclTable    &newT,
+                                AclTable    &curT,
+                                set<string> &addSet,
+                                set<string> &delSet)
 {
+    set<string> newPortSet, curPortSet;
+
+    // Collect new ports
+    for (auto p : newT.pendingPortSet)
+    {
+        newPortSet.insert(p);
+    }
+    for (auto p : newT.portSet)
+    {
+        newPortSet.insert(p);
+    }
+
+    // Collect current ports
+    for (auto p : curT.pendingPortSet)
+    {
+        curPortSet.insert(p);
+    }
+    for (auto p : curT.portSet)
+    {
+        curPortSet.insert(p);
+    }
+
+    // Get the difference newPortSet-curPortSet and curPortSet-newPortsSet
+    std::set_difference(newPortSet.begin(), newPortSet.end(),
+                        curPortSet.begin(), curPortSet.end(),
+                        std::inserter(addSet, addSet.end()));
+    std::set_difference(curPortSet.begin(), curPortSet.end(),
+                        newPortSet.begin(), newPortSet.end(),
+                        std::inserter(delSet, delSet.end()));
+
+}
+
+bool AclOrch::updateAclTablePorts(AclTable &newTable, AclTable &curTable)
+{
+    sai_object_id_t    port_oid = SAI_NULL_OBJECT_ID;
+    set<string>        addPortSet, deletePortSet;
+
     SWSS_LOG_ENTER();
+    getAddDeletePorts(newTable, curTable, addPortSet, deletePortSet);
 
+    // Lets first unbind and unlink ports to be removed
```

Comment:
Not really, especially for Dynamic Port Breakout feature, we will never have both. First will remove ports, and then add ports. But if some one tries to add and delete ports in one shot, order shouldn't matter as the both sets are mutually exclusive. Also, note that ACL table can exist with empty port list, otherwise we would have been forced to add first!